hardware_interface/src/resource_manager.cpp

```diff
@@ -756,50 +793,83 @@ ResourceManager::ResourceManager(
 }
 
 // CM API: Called in "callback/slow"-thread
-void ResourceManager::load_urdf(
-  const std::string & urdf, bool validate_interfaces, bool load_and_initialize_components)
+bool ResourceManager::load_and_initialize_components(const std::string & urdf)
 {
-  is_urdf_loaded__ = true;
+  components_are_loaded_and_initialized_ = true;
+
+  const auto hardware_info = hardware_interface::parse_control_resources_from_urdf(urdf);
+
```

Comment:
```suggestion
  const auto hardware_info = hardware_interface::parse_control_resources_from_urdf(urdf);
  if(
    !resource_storage_->actuators_.empty() || !resource_storage_->sensors_.empty() ||
    !resource_storage_->systems_.empty())
  {
     RCUTILS_LOG_WARN_NAMED(
        "resource_manager", "It seems the resource manager's components are already loaded and initialized! Skipping!");
      return;
  }
```

Maybe something like this will fix the above issue?

Reply:
I think we don't have an issue, but not 100% sure

Reply:
I was suggesting this as a fix for the case of Gazebo functionality, if it can be fixed in a different way. I'm okay with it

Reply:
Gazebo now has also compatible changes with this. see ros-controls/gz_ros2_control#265